### PR TITLE
refactor genomepy rules

### PR DESCRIPTION
**Pull request by siebrenf**
**What problem is the PR solving / What's new?**
genomepy related rules are doing too much at once, giving snakemake a headache.

**What did change**
All genomepy rules have been refactored. Each file is now downloaded separately, and (mostly) through the front end API (I'll try to make the front end more amenable to this too in the future).

Most importantly, the blacklist is now downloaded separately. If genomepy cannot download a blacklist because it does not exist, the new rule outputs an empty file.

**Checklist**
- [x] I made a PR to develop (not master)
- [ ] If applicable: I updated the docs
- [ ] I updated the CHANGELOG
- [ ] These changes are covered by the tests

### CODE REVIEW

File: seq2science/scripts/get_genome_annotation.py [hunk 1]
```diff
@@ -1,41 +1,50 @@
...
 
 
 with open(snakemake.log[0], "w") as log:
     with contextlib.redirect_stdout(log), contextlib.redirect_stderr(log):
-        try:
-            provider = snakemake.params.providers[snakemake.wildcards.raw_assembly]["annotation"]
+        # list user plugins
+        active_plugins = genomepy.functions.config.get("plugin", [])
+        # deactivate user plugins
+        genomepy.funcions.manage_plugins("disable", active_plugins)
```
Comment (Maarten-vd-Sande) [inline seq2science/scripts/get_genome_annotation.py:14]: This breaks when doing this in parallel, not that I really care tbf :innocent:

Comment (Maarten-vd-Sande) [reply seq2science/scripts/get_genome_annotation.py:14]: If we want to do this properly, this should come from Snakemake. And onexit and onerror these should be set back. Perhaps even easier to do the whole thing there?

Comment (Maarten-vd-Sande) [reply seq2science/scripts/get_genome_annotation.py:14]: Somewhere here, although that is super confusing this is in loggen :yum: 

https://github.com/vanheeringen-lab/seq2science/blob/develop/seq2science/rules/configuration_logging.smk#L7-L22

Comment (Maarten-vd-Sande) [reply seq2science/scripts/get_genome_annotation.py:14]: Anyways, I don't think it really matters

Comment (siebrenf) [reply seq2science/scripts/get_genome_annotation.py]: all genomepy rules share resource `parallel_downloads=1`, so there should be no parallel genomepy runs.

Comment (Maarten-vd-Sande) [reply seq2science/scripts/get_genome_annotation.py]: parallel_downloads is set to three though :upside_down_face:

Comment (Maarten-vd-Sande) [reply seq2science/scripts/get_genome_annotation.py]: (in that three downloads run in parallel0

Comment (Maarten-vd-Sande) [reply seq2science/scripts/get_genome_annotation.py]: This is to bypass the problem when downloading e.g. two samples, where one is five time the size of the other. This guarantees a more continuous stream of files being downloaded for downstream processing. Anyways, I don't think we really have to worry about these cases

Comment (siebrenf) [reply seq2science/scripts/get_genome_annotation.py]: its an edge case yea. I've added a genomepy downloads resource too. should be fixed forever now :)

**Pull request merged**